### DISCUSSION TITLE
How to hollow out a mesh with inconsistent surface normals and internal normals?

### DISCUSSION THREAD

**Discussion by NamTran838P** [pyvista/pyvista#4797] [Q&A] [answered]
Hi PyVista community,

I would like to hollow out a mesh. It can be downloaded at:

https://drive.google.com/file/d/1sVNJjyKG9Ptr1NT84QJGwICSkZT5AEEQ/view?usp=sharing

This mesh is not hollow. It contains internal vertices and faces at the smaller end. Using MeshLab, it is possible to zoom in and view these internal faces at the smaller end. Therefore, I encountered some difficulties trying to hollow out this mesh. I have tried the following methods: 

1) Using PyVista to subtract the offset mesh: https://github.com/pyvista/pyvista/discussions/3121
2) Using MeshLab to create an offset mesh with uniform mesh resampling, flip its normals and merge with original mesh.

Both attempts have been unsuccessful. How can I hollow out my mesh?

Thanks a lot!

**Comment by darikg** [comment, marked as answer]
I think you want something like [3d alpha wrapping](https://doc.cgal.org/latest/Alpha_wrap_3/index.html) in CGAL, or [blender's shrinkwrap modifier](https://docs.blender.org/manual/en/latest/modeling/modifiers/deform/shrinkwrap.html). I don't think VTK / pyvista has a similar feature.
